csim.c

```diff
@@ -9,10 +9,41 @@
 #include <stdio.h>
 #include <stdlib.h>
 
-void clear_cache(struct Cache *cache) {
-	if(cache != NULL){
-		//clear
+cache build_cache(long int numSets, int numLines, long int blockSize){
+	cache newCache;	
+	cacheSet set;
+	setLine line;
```

Comment:
I think I named this CacheLine, also shouldn't there only be newCache declared here?